### PR TITLE
test: Update fedora-atomic to fedora 23

### PR DESCRIPTION
**Pull request by petervo**

### CODE REVIEW

File: test/check-multi-machine [hunk 1]
```diff
@@ -405,25 +406,27 @@ class TestMultiMachine(MachineCase):
...
-        b.wait_in_text('#troubleshoot-dialog h4', "Could not contact")
-        b.wait_present("#edit-machine-port")
-        b.set_val("#edit-machine-port", "2222")
-        b.click('#troubleshoot-dialog .btn-primary')
-        b.wait_in_text('#troubleshoot-dialog h4', "Log in to")
-        b.wait_visible("#login-diff-password")
-        b.set_val("#login-custom-password", "alt-password")
-        b.click('#troubleshoot-dialog .btn-primary')
-        b.wait_popdown('troubleshoot-dialog')
+        # https://bugzilla.redhat.com/show_bug.cgi?id=1290659
```
Comment (stefwalter) [inline test/check-multi-machine]: Perhaps this should be a known issue with fedora-atomic. It's something that actually affects users (every time they try to perform the task of changing the ssh port), and something Cockpit cannot work around.

File: test/check-system-info [hunk 2]
```diff
@@ -82,7 +82,7 @@ class TestSystemInfo(MachineCase):
         b = self.browser
 
         network_time_prefix = "NTP enabled"
-        if m.image == "fedora-23" or m.image == "fedora-testing":
+        if m.image == "fedora-atomic" or m.image == "fedora-23" or m.image == "fedora-testing":
```
Comment (stefwalter) [inline test/check-system-info]: Can we start using 'in [x, y, z]' here?

**Comment by stefwalter** [comment]
I really don't like that we're missing a lot of bugs due to skips in our tests in atomic. But lets try to solve that in another bug.

**Comment by stefwalter** [comment]
Should we remove this line after this update?

```
test/check-multi-machine-key:@unittest.skipIf("atomic" in os.environ.get("TEST_OS", "") or "rhel" in os.environ.get("TEST_OS", ""), "key auth: not supported")
```

**Comment by petervo** [comment]
> Should we remove this line after this update?
>
> |test/check-multi-machine-key:@unittest.skipIf("atomic" in
> os.environ.get("TEST_OS", "") or "rhel" in os.environ.get("TEST_OS",
> ""), "key auth: not supported") |
>

This won't work out of the box on atomic because the ssh pam stack is 
used instead of the cockpit pam stack.

Do we want to change the sshd pam stack so the test can run?

**Comment by stefwalter** [comment]
> This won't work out of the box on atomic because the ssh pam stack is used instead of the cockpit pam stack.

Isn't that just one part of the test? The test could still load the keys manually, the way a user would, right? Obviously that part could be conditional. But I guess this should be another bug.

**Comment by petervo** [comment]
Updated the check-multi-machine-key test to run on atomic.

@mvollmer does the check networking failure look familiar to you at all? It's running the same version of network manager as fedora23 so i'm not sure what is going on.

**Comment by stefwalter** [comment]
@petervo Could you include this commit?

```
diff --git a/test/guest/fedora-atomic.setup b/test/guest/fedora-atomic.setup
index 83e8860..625bd22 100755
--- a/test/guest/fedora-atomic.setup
+++ b/test/guest/fedora-atomic.setup
@@ -32,8 +32,8 @@ systemctl start docker || journalctl -u docker
 docker pull cockpit/ws
 docker pull busybox:latest
 docker pull busybox:buildroot-2014.02
-docker pull submod/helloapache
-docker pull submod/atomicapp
+docker pull submod/helloapache:0.1.11
+docker pull gcr.io/google_containers/pause:0.8.0
 
 # Final tweaks
 
```

**Comment by petervo** [comment]
Added the container changes and split the multi-machine-key tests into a separate PR.

File: test/check-system-info [hunk 3]
```diff
@@ -82,7 +82,7 @@ class TestSystemInfo(MachineCase):
         b = self.browser
 
         network_time_prefix = "NTP enabled"
-        if m.image == "fedora-23" or m.image == "fedora-testing":
+        if m.image in {"fedora-atomic", "fedora-23", "fedora-testing"}:
```
Comment (mvollmer) [inline test/check-system-info]: What about doing it like this: https://github.com/mvollmer/cockpit/commit/ce81ae0a28142422656c4a5dc1bec73b6178695d

**Comment by mvollmer** [comment]
I think the check-networking issue goes like this:

- The main (and only) network interface when booting any of our images is called ens2.

- Images except fedora-atomic have ifcfg-ens2 as expected, with DEVICE=ens2 in it.  I don't know who has written that file.

- The fedora-atomic image has ifcfg-ens3, with DEVICE=ens3 in it.  I don't know who has written that file, either. The main and only interface is still ens2, so ifcfg-ens3 gets ignored and NM creates a ad-hoc, unsaved configuration for ens2.

- In the test, we hotplug a new interface, it gets called ens3 (haven't checked that yet), and then we have two connection settings that apply to it, the old ifcfg-ens3 one, and the new one that the test has written with HWADDR=... in it.

- Cockpit tries to pick the most recently used one, but it's not clear which one that is going to be.

So the bug is that there is a stale ifcfg-ens3 in the fedora-atomic image.  I think it would be good to figure out where it comes from.

As a fix, we could just remove ifcfg-ens3 explicitly in the test or during image building.  That might be more direct and less intrusive than changing the naming scheme of all interfaces.

Alternatively, we could explicitly activate the right connection via "nmcli c up test" and then check that Cockpit picks up the right one.

Cockpit is supposed to work with multiple connection settings for a single interface, although it doesn't admit in the UI that there is more than one.  It just works on the most recently used one.
So we might even explicitly create multiple connection settings.

(More later.)

**Comment by mvollmer** [comment]
> Images except fedora-atomic have ifcfg-ens2 as expected, with DEVICE=ens2 in it.  I don't know who has written that file.

Ahem:

```
# Generated by dracut initrd
NAME="ens2"
DEVICE="ens2"
...
```

**Comment by mvollmer** [comment]
> In the test, we hotplug a new interface, it gets called ens3 (haven't checked that yet)

It is actually called `eth0`, which surprises me.

File: test/guest/fedora-atomic.setup [hunk 4]
```diff
@@ -18,12 +18,17 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
 
+# Use mac's in name policy, prevents interference from network-scripts
```
Comment (mvollmer) [inline test/guest/fedora-atomic.setup]: I propose this change to check-networking instead:

```
diff --git a/test/check-networking b/test/check-networking
index b34b06c..4771879 100755
--- a/test/check-networking
+++ b/test/check-networking
@@ -117,6 +117,10 @@ class TestNetworking(MachineCase):
         m.execute("nmcli c reload")
         iface = self.add_iface(mac)
 
+        # Activate our settings on the new interface
+        wait(lambda: m.execute('nmcli d | grep %s | grep -v unavailable' % iface))
+        m.execute('nmcli c up "System %s"' % label)
+
         self.wait_for_iface(iface)
         b.click("#networking-interfaces tr[data-interface='%s']" % iface)
         b.wait_visible("#network-interface")
```

This makes the expected state more explicit.

Comment (mvollmer) [inline test/guest/fedora-atomic.setup]: However, we will never receive any metrics for the interfaces, so wait_for_iface will fail.  Do we expect metrics to work on Atomic?  I'll look into this.

Comment (mvollmer) [inline test/guest/fedora-atomic.setup]: My objection is that I'd rather make the test more robust than tweak the OS to make a fragile test happy.  Thus, we should probably explicitly create two settings for the same mac, activate one, and check that Cockpit shows the correct one.

However, since this means we also need to figure out the metrics issue, it's probably better to do all this in another PR.

Comment (mvollmer) [inline test/guest/fedora-atomic.setup]: > However, since this means we also need to figure out the metrics issue, it's probably better to do all this in another PR.

The issue is that, *drumroll*, we don't have any internal metrics for individual network interfaces. We only have metrics for the total traffic.

Thus without PCP, we don't get any data for the graphs etc on the networking pages.  The tests are not normally affected by this, but my proposed change above means that the interface is already past the "Configuring IP" stage when we call wait_for_iface, and wait_for_iface thus waits for some traffic rate to appear, which doesn't.

Comment (mvollmer) [inline test/guest/fedora-atomic.setup]: So, objections temoprarily withdrawn, let's go with @petervo's version and I make a separate PR.

Comment (mvollmer) [inline test/guest/fedora-atomic.setup]: Files followups for me: #3338, #3339
